blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricIO.java

```diff
@@ -254,4 +254,14 @@ protected void addRollupToBatch(BatchStatement batch, Locator locator,
         }
         return locatorTimestampRollup;
     }
+
+    /**
+     * Retrieves the {@link BoundStatement} for a particular metric and granularity.
+     * Subclasses will implement this.
+     *
+     * @param metric
+     * @param granularity
+     * @return
+     */
+    protected abstract BoundStatement getBoundStatementForMetric(IMetric metric, Granularity granularity);
```

Comment:
I can't think of a useful way to provide default implementation here. If you look at the implementation of this method in the sub classes, each line requires knowledge of what object (BluefloodSetRollup, BasicRollup, etc) you are operating on. I briefly thought of using generic types but decided it's not worth it.

Do you have specific suggestions?